src/Xenial.Tasty.Cli/Views/MainWindow.ViewModel.cs

```diff
@@ -24,24 +26,32 @@ internal class MainWindowViewModel : IDisposable
         public int SeparatorSize { get; set; } = 100;
         internal Terminal.Gui.ColorScheme ColorScheme { get; private set; } = null!; //Trick the compiler for SetColor method
         internal string ColorSchemeName { get; private set; } = null!; //Trick the compiler for SetColor method
-        internal TastyCommander Commander { get; }
+        internal TastyProcessCommander Commander { get; }
         internal string LogText { get; private set; } = string.Empty;
-        internal Progress<(string line, bool isRunning, int exitCode)> LogProgress { get; }
+        internal string CurrentProject { get; private set; } = string.Empty;
+
+        internal Progress<(string line, bool isError, int? exitCode)> LogProgress { get; }
         internal ObservableCollection<CommandItem> Commands { get; } = new ObservableCollection<CommandItem>();
         private CancellationTokenSource CancellationTokenSource { get; } = new CancellationTokenSource();
 
         public MainWindowViewModel(string colorSchemeName, Terminal.Gui.ColorScheme colorScheme)
         {
-            Commander = new TastyCommander()
-                .UseNamedPipesTransport()
-                .RegisterReporter(Report)
-                .RegisterReporter(ReportSummary);
+            var connectionString = NamedPipesConnectionStringBuilder.CreateNewConnection();
 
-            LogProgress = new Progress<(string line, bool isRunning, int exitCode)>(p =>
+            LogProgress = new Progress<(string line, bool isError, int? exitCode)>(p =>
             {
                 LogText += $"{p.line.TrimEnd(Environment.NewLine.ToArray())}{Environment.NewLine}";
```

Comment:
StringBuilder?

Reply:
So the StringBuilder will go away anyway